### PR TITLE
Upgrade C# testing images to debian stretch

### PR DESCRIPTION
**Pull request by jtattermusch**
- stretch is now the stable debian distro
- .NET core 2+ (now the stable version) doesn't support jessie (it supports stretch and newer) and we should switch to using .NET core 2.x SDK soon as it is desirable to use an up-to-date SDK.
- this PR has no influence on platforms that gRPC C# actually supports, only allows us to use newer  SDK for build.
- .NET core 2.x is necessary to build netstandard2.0 and netcoreapp2.1 targets.

### CODE REVIEW

File: templates/tools/dockerfile/csharp_dotnetcli_deps.include [hunk 1]
```diff
@@ -1,12 +1,38 @@
 # Install dotnet SDK based on https://www.microsoft.com/net/core#debian
-RUN apt-get update && apt-get install -y curl libunwind8 gettext
+#RUN apt-get update && apt-get install -y curl libunwind8 gettext
```
Comment (apolcyn) [inline templates/tools/dockerfile/csharp_dotnetcli_deps.include]: nit: delete rather than uncomment these lines?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,021,324      Total (=)      2,021,324

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,177,662      Total (>)     11,177,657

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/15599
https://github.com/grpc/grpc/issues/15308

**Comment by jtattermusch** [comment]
oauth2_auth_token on CoreCLR seems to be failing: https://source.cloud.google.com/results/invocations/a816bca0-272b-4fbb-835e-6562f48b08e7/targets/github%2Fgrpc%2Finterop_test/tests;group=tests;test=cloud_to_prod_auth:csharpcoreclr:default:oauth2_auth_token:tls;row=1

**Comment by jtattermusch** [comment]
the interop test failure (https://source.cloud.google.com/results/invocations/a816bca0-272b-4fbb-835e-6562f48b08e7) seems to be due to a broken dotnet 1.0.x installation on debian stretch, which is hard to fix.

superseded by https://github.com/grpc/grpc/pull/17309.

**Pull request closed without merging**